dynamic_instrumentation/src/instrument_memory.rs

```diff
@@ -512,32 +514,56 @@ impl<'a, 'tcx: 'a> Visitor<'tcx> for CollectFunctionInstrumentationPoints<'a, 't
                         },
                     );
                 }
-            } else if value.ty(&self.body.local_decls, self.tcx).is_region_ptr() {
-                /*
-                    Stephen: uncomment below to get an example of following error:
-
-                    error: internal compiler error: compiler/rustc_borrowck/src/borrow_set.rs:252:17:
-                    found two uses for 2-phase borrow temporary _9: bb45[0] and bb46[1]
-                    --> ../analysis/test/src/pointers.rs:162:16
-                        |
-                    162 |     for arg in ::std::env::args() {
-                        |                ^^^^^^^^^^^^^^^^^^
-                */
-
-                // location.statement_index += 1;
-                // if let Some(p) = rv_place(value) {
-                //     self.add_instrumentation_point(
-                //         location,
-                //         ref_copy_fn,
-                //         vec![Operand::Copy(dest)],
-                //         false,
-                //         false,
-                //         EventMetadata {
-                //             source: Some(to_mir_place(&p)),
-                //             destination: Some(to_mir_place(&dest)),
-                //         },
-                //     );
-                // }
+            } else if value.ty(&self.body.local_decls, self.tcx).is_region_ptr() { // Rhs is a ref
+                /// Used to strip initital deref from projection sequences
+                fn pop_one_projection<'tcx>(p: &Place<'tcx>, tcx: TyCtxt<'tcx>) -> Option<Place<'tcx>> {
+                    if p.projection.is_empty() {
+                        return None;
+                    }
+                    Some(Place {
+                        local: p.local,
+                        projection: tcx.intern_place_elems(&p.projection[..p.projection.len() - 1]),
+                    })
+                }
+                if let Rvalue::Ref(_, bkind, p) = value {
```

Comment:
this looks good to me, but I think it only covers cases like
```rust
statement: _7 = &mut (*_6)
```
and not something like this on the hypothetical next line
```rust
statement: _8 = move _7
```
I think it's fine to add this support to a subsequent PR.

Other cases that might need attention are storing to a reference's field, loading a reference's fields, etc. Perhaps we can combine the eventual work on address-taken locals with what we need to do here.